### PR TITLE
Created example IP with one rep (SIARD2 database)

### PR DESCRIPTION
**Pull request by PhillipTommerholt**
Created example IP with one rep (SIARD2 database)

### CODE REVIEW

**Review by koit — CHANGES_REQUESTED** [review]
Overall it looks very good. I proposed some minor changes to root METS.xml in the branch `IP_1rep_SIARD2_fixes`, see PR #213 (merge it to get my changes into `corpus/example/IP_1rep_SIARD2` branch). Regarding representations/rep1/METS.xml, I have the following comments:

- Line 11: `mets/@TYPE="Database"` - these values should come from the vocabulary ["VocabularyType"](http://earkcsip.dilcis.eu/schema/DILCISVocabulariesIP.xml) and the right one is in plural `Databases`.
- Line 34-36: `dmdSec` referring to ead.xml is in both mets files - I'd keep only one of these. In an IP with only one rep it doesn't matter much which way to choose, but I would avoid duplicate references as these can create logical loops that might cause problems.
- Line 34, 38, 42 etc. - all xsd:ID attribute values have to be unique within the IP (CSIP18, CSIP33, CSIP46 etc.). I'd add prefixes to representation METS.xml IDs, e.g. instead of "ID-dmdsecID" use "ID-rep1-mets-dmdsecID"
- Line 35: `xlink:href="metadata/descriptive/ead.xml"` - this is not 100% clear, as we discussed today, but I'd use `xlink:href="../../metadata/descriptive/ead.xml"`. Or maybe leave it now and wait for the official decision that should be:
   a) all paths are relative to the current file, or 
   b) all paths are absolute from the root of the package.

When you're done with the representation METS.xml, don't forget to update the `file/@CHECKSUM` and `file/@SIZE` on line 92 of root mets file.

**Review by koit — APPROVED** [review]

**Review by koit — CHANGES_REQUESTED** [review]
The issues from my last review have been fixed, but I found some new ones.

### 1. Root METS.xml

**92:** rep METS.xml has changed since you last updated md5 and size, the current values are 43edc6d2278c0fbccbe62627e21d31ff and 11594:

```xml
<file  ID="ID-rep1" MIMETYPE="application/xml" CHECKSUM="bb067c530ab7682b2be175dadfd18bbb" CHECKSUMTYPE="MD5" SIZE="11760">
```

**113:** `div/@CONTENTIDS` has bad ID, should be "ID-fileGrp_schemas":

```xml
<div ID="ID-Structmap_Div_ID_Schemas" LABEL="Schemas" CONTENTIDS="ID-IP_with_dmd_fileGrp_schemas">
```

fileGrp ID is:

```xml
<fileGrp USE="Schemas" ID="ID-fileGrp_schemas">
```

### Representation METS.xml

**79:** `fileSec/fileGrp/file` for `representations/rep1/schemas/ead3.xsd` was deleted, but the file is still present in the folder:

```diff
-   <!-- ead3.xsd -->
-   <file ID="ID-IP_with_dmd_fileGrp_schemas_ead3.xsd" MIMETYPE="application/xml"  CREATED="2016-12-14T14:20:00" 
-     CHECKSUM="7619e9dc6d1b592238e04a1d9be49c06" CHECKSUMTYPE="MD5" SIZE="92510">
-     <FLocat LOCTYPE="URL" xlink:href="schemas/ead3.xsd" xlink:type="simple"/>
-   </file>
```

**154:** typo in comment text "vocabualry"

```xml
<!-- CSIP81 mets/structMap/@TYPE The type attribute of the structural map (structMap) is set to value “PHYSICAL” from the vocabualry -->
```

**144, 153, 155, 157:** `div/@ID` value not unique across IP (is the same as in root METS.xml):

```xml
<structMap TYPE="PHYSICAL" LABEL="CSIP StructMap" ID="ID-StructmapID"> 
/…/
    <div ID="ID-Structmap_Div_ID_Metadata" LABEL="Metadata">
    </div>
    <div ID="ID-Structmap_Div_ID_Documentation" LABEL="Documentation">
    </div>
    <div ID="ID-Structmap_Div_ID_Schemas" LABEL="Schemas" CONTENTIDS="IP_with_database_fileGrp_schemas">
```

**157:** `div/@CONTENTIDS` prefix "ID-" missing:

```xml
<div ID="ID-Structmap_Div_ID_Schemas" LABEL="Schemas" CONTENTIDS="IP_with_database_fileGrp_schemas">
```

vs:

```xml
<fileGrp USE="Schemas" ID="ID-IP_with_database_fileGrp_schemas">
```

**159:** Missing `div/@CONTENTIDS` reference (CSIP108 or CSIP104), should be:

```xml
<div ID="ID-Structmap_Div_ID_data" LABEL="Data" CONTENTIDS="ID-IP_with_fileSec_fileSec">
```

**Comment by PhillipTommerholt** [comment]
Thanks for your review @koit !

I don't understand the first one.

> ### 1. Root METS.xml
> **92:** rep METS.xml has changed since you last updated md5 and size, the current values are 43edc6d2278c0fbccbe62627e21d31ff and 11594:
> 
> ```
> <file  ID="ID-rep1" MIMETYPE="application/xml" CHECKSUM="bb067c530ab7682b2be175dadfd18bbb" CHECKSUMTYPE="MD5" SIZE="11760">
> ```

Are you sure about this? I double checked and this seems to be the correct MD5-sum and size...

**Review by andersbonielsen — COMMENTED** [review]
I went through the comments Koit made as well as the changes Phillip made afterwards.
The example seems to be correct.

File: corpus/template-ip/IP_1rep_SIARD2/representations/rep1/METS.xml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        </file>
+         <file  ID="ID-JPEG17" MIMETYPE="image/jpeg" CREATED="2019-04-16T12:10:00" CHECKSUM="40a5309f5432d87cc9004ab4de65c868" CHECKSUMTYPE="MD5" SIZE="12203">
+          <FLocat LOCTYPE="URL" xlink:type="simple" xlink:href="data/northwind.siard_lobseg_1/content/schema1/table5/lob15/record9.jpg" /> 
+        </file>
+      
+      </fileGrp>
+     
+  </fileSec>
+  
+  <structMap TYPE="PHYSICAL" LABEL="CSIP StructMap" ID="ID-StructmapID">
```
Comment (koit) [inline corpus/template-ip/IP_1rep_SIARD2/representations/rep1/METS.xml]:
```suggestion
  <structMap TYPE="PHYSICAL" LABEL="CSIP StructMap" ID="ID-StructmapID_rep1"> 
```

The ID collides with the root METS.xml.

**Comment by koit** [comment]
Regarding md5 and file size: I couldn't figure out where our differences come from - we must be somehow using different versions of the file.

For example, if I take the `representations/rep1/METS.xml` at your latest commit c35ff707cca793665334e1d59185216c2fb96415 (either by checking it out to my local repo or by clicking "Raw" and "Save As" at [this link](https://github.com/DILCISBoard/eark-ip-test-corpus/blob/c35ff707cca793665334e1d59185216c2fb96415/corpus/template-ip/IP_1rep_SIARD2/representations/rep1/METS.xml)), the md5 value is 21b5a81c270320eef062e7345fcf86f5 and size is 11651.

I thought the end-of-line symbols might get changed, as I'm using the Unix LF (ASCII x0a), while Windows default is CRLF (ASCII x0d0a), but it doesn't seem to be so, because all the other files in this IP also use the Unix standard and we agree on their md5.

But in general: except for the `structMap/@ID` in rep METS.xml and after that, the update to root METS.xml regarding rep METS.xml md5 and size, the package seems perfect.

**Pull request closed without merging**